src/Symfony/Component/Notifier/Transport/FailoverTransport.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Notifier\Transport;
+
+use Symfony\Component\Notifier\Message\MessageInterface;
+
+/**
+ * Uses several Transports using a failover algorithm.
+ *
+ * @author Fabien Potencier <fabien@symfony.com>
+ *
+ * @experimental in 5.0
+ */
+class FailoverTransport extends RoundRobinTransport
```

Comment:
is this legit? `FailoverTransport instanceof RoundRobinTransport`?
should the common logic be moved to a trait instead, or an abstract class?